signature/jws/algorithm.go

```diff
@@ -47,3 +48,37 @@ func SigningMethodFromKey(key interface{}) (jwt.SigningMethod, error) {
 	}
 	return nil, errors.New("key not recognized")
 }
+
+func keySpecFromKey(key interface{}) (notation.KeySpec, error) {
+	if k, ok := key.(interface {
+		Public() crypto.PublicKey
+	}); ok {
+		key = k.Public()
+	}
+
+	switch key := key.(type) {
+	case *rsa.PublicKey:
+		switch key.Size() {
+		case 256:
+			return notation.RSA_2048, nil
+		case 384:
+			return notation.RSA_3072, nil
+		case 512:
+			return notation.RSA_4096, nil
+		default:
+			return notation.RSA_2048, nil
```

Comment:
We should error out on unsupported key sizes.

```suggestion
		default:
			return "", fmt.Errorf("RSA key of size %q bits is not supported", key.Size()*8)
```